module/input/SensorFilter/src/SensorFilter.cpp

```diff
@@ -538,11 +540,30 @@ namespace module::input {
                                 feet_down[BodySide::RIGHT] = true;
                                 feet_down[BodySide::LEFT]  = true;
                             }
+                        }
+                        else if (config.footDown.method() == "FSR") {
+                            // For a foot to be on the ground we want a minimum of 2 diagonally opposite studs
+                            // in contact with the ground
+                            // So fsr1 and fsr3, or fsr2 and fsr4
+                            //
+                            // A FSR is in contact with the ground if its value is greater than the certainty threshold
+
+                            feet_down[BodySide::LEFT] = (((input.fsr.left.fsr1 > config.footDown.threshold())
+                                                          && (input.fsr.left.fsr3 > config.footDown.threshold()))
+                                                         || ((input.fsr.left.fsr2 > config.footDown.threshold())
+                                                             && (input.fsr.left.fsr4 > config.footDown.threshold())));
+
+                            feet_down[BodySide::RIGHT] = (((input.fsr.right.fsr1 > config.footDown.threshold())
+                                                           && (input.fsr.right.fsr3 > config.footDown.threshold()))
+                                                          || ((input.fsr.right.fsr2 > config.footDown.threshold())
+                                                              && (input.fsr.right.fsr4 > config.footDown.threshold())));
+                        }
 
-                            if (this->config.debug) {
-                                emit(graph("Sensor/Foot Down/Z/Left", feet_down[BodySide::LEFT]));
-                                emit(graph("Sensor/Foot Down/Z/Right", feet_down[BodySide::RIGHT]));
-                            }
+                        if (this->config.debug) {
```

Comment:
Is this the normal way we do this? I thought on the webots module we were choosing whether to log things by using `log<NUClear::${log_type}>(..)`
I might be misunderstanding things a bit here.

Reply:
This is guarding emits, not logs?

Reply:
Good point